src/Microsoft.AspNet.Mvc.Core/ModelBinding/Metadata/DisplayMetadata.cs

```diff
@@ -40,10 +41,10 @@ public class DisplayMetadata
         public string DisplayFormatString { get; set; }
 
         /// <summary>
-        /// Gets or sets a display name for the model.
+        /// Gets or sets a delegate used for getting the display name for the model.
         /// See <see cref="ModelMetadata.DisplayName"/>
         /// </summary>
-        public string DisplayName { get; set; }
+        public Func<string, string> DisplayName { get; set; }
```

Comment:
Well, I agree that it is in property specific context ( when the metadata represents a property), Having a string key as an input enables cases when someone wants to use it as a key to lookup a resource and this key might not be a property name ( for example when the metadata represents a type). Unless I hear a strong push against it this stays as is.